addons/l10n_tw_edi_ecpay/models/account_move.py

```diff
@@ -0,0 +1,280 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import base64
+import datetime
+import uuid
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.addons.l10n_tw_edi_ecpay.utils import EcPayAPI
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    # ------------------
+    # Fields declaration
+    # ------------------
+
+    l10n_tw_edi_file_id = fields.Many2one(
+        comodel_name="ir.attachment",
+        compute=lambda self: self._compute_linked_attachment_id("l10n_tw_edi_file_id", "l10n_tw_edi_file"),
+        depends=["l10n_tw_edi_file"],
+        copy=False,
+        readonly=True,
+        export_string_translation=False,
+    )
+    l10n_tw_edi_file = fields.Binary(
+        string="Ecpay JSON File",
+        copy=False,
+        readonly=True,
+        export_string_translation=False,
+    )
+    l10n_tw_edi_ecpay_invoice_id = fields.Char(string="Ecpay Invoice Number", readonly=True, copy=False)
+    l10n_tw_edi_related_number = fields.Char("Related Number", readonly=True, copy=False)
+    # False => Not sent yet.
+    l10n_tw_edi_state = fields.Selection(
+        selection=[("invoiced", "Invoiced"), ("valid", "Valid"), ("invalid", "Invalid")],
+        string="Invoice Status",
+        copy=False,
+        readonly=True,
+        tracking=True,
+    )
+    l10n_tw_edi_ecpay_is_included_tax = fields.Selection(
+        selection=[("1", "tax included"), ("0", "untax")],
+        string="Included Tax",
+        compute="_compute_is_included_tax",
+    )
+    l10n_tw_edi_is_donate = fields.Boolean(string="Donate", copy=False, compute="_compute_is_donate", store=True, readonly=False)
+    l10n_tw_edi_love_code = fields.Char(string="Love Code", copy=False, compute="_compute_is_donate", store=True, readonly=False)
+    l10n_tw_edi_is_print = fields.Boolean(string="Get Printed Version", copy=False, compute="_compute_is_print", store=True, readonly=False)
+    l10n_tw_edi_carrier_type = fields.Selection(
+        selection=[("1", "ECpay e-invoice carrier"), ("2", "Citizen Digital Certificate"), ("3", "Mobile Barcode")],
+        string="Carrier Type",
+        copy=False,
+        compute="_compute_carrier_info",
+        store=True,
+        readonly=False)
+    l10n_tw_edi_carrier_number = fields.Char(string="Carrier Number", copy=False, compute="_compute_carrier_info", store=True, readonly=False)
+    l10n_tw_edi_invoice_amount = fields.Float(string="Invoice Amount", readonly=True, copy=False)
+    l10n_tw_edi_invoice_shop_custom_number = fields.Char(string="Cooperative Shop Custom Number", readonly=True, copy=False)
+    l10n_tw_edi_invoice_create_date = fields.Datetime(string="Creation Date", readonly=True, copy=False)
+    l10n_tw_edi_refund_finish = fields.Boolean(string="Refund Finish", readonly=True, copy=False)
```

Comment:
This is not needed, just check the state if you need to see if the refund is done or not.